File: app/src/main/java/com/summer/itis/summerproject/model/repository/RepositoryProvider.java
package com.summer.itis.summerproject.model.repository;

import com.summer.itis.summerproject.model.repository.json.CardRepository;
import com.summer.itis.summerproject.model.repository.json.QuestionRepository;
import com.summer.itis.summerproject.model.repository.json.TestRepository;
import com.summer.itis.summerproject.model.repository.json.UserRepository;


public class RepositoryProvider {

    private static UserRepository userRepository;

    private static TestRepository testRepository;

    private static CardRepository cardRepository;

    private static QuestionRepository questionRepository;

    public static UserRepository getUserRepository() {
        if (userRepository == null) {
            userRepository = new UserRepository();
        }
        return userRepository;
    }

    public static void setUserRepository(UserRepository userRepository) {
        RepositoryProvider.userRepository = userRepository;
    }

    public static TestRepository getTestRepository() {
        if (testRepository == null) {
            testRepository = new TestRepository();
        }
        return testRepository;
    }

    public static void setTestRepository(TestRepository testRepository) {
        RepositoryProvider.testRepository = testRepository;
    }

    public static CardRepository getCardRepository() {
        if (cardRepository == null) {
            cardRepository = new CardRepository();
        }
        return cardRepository;
    }

    public static void setCardRepository(CardRepository cardRepository) {
        RepositoryProvider.cardRepository = cardRepository;
    }

    public static QuestionRepository getQuestionRepository() {
        if (questionRepository == null) {
            questionRepository = new QuestionRepository();
        }
        return questionRepository;
    }

    public static void setQuestionRepository(QuestionRepository questionRepository) {
        RepositoryProvider.questionRepository = questionRepository;
    }
}

File: app/src/main/java/com/summer/itis/summerproject/repository/RepositoryProvider.kt
package com.summer.itis.summerproject.repository

import android.util.Log
import com.summer.itis.summerproject.repository.api.WikiApiRepository
import com.summer.itis.summerproject.repository.api.WikiApiRepositoryImpl
import com.summer.itis.summerproject.repository.json.*
import com.summer.itis.summerproject.utils.Const.TAG_LOG


class RepositoryProvider {

    companion object {
        val testRepository: TestRepository by lazy {
            TestRepository()
        }

        val cardRepository: CardRepository by lazy {
            CardRepository()
        }

        val userRepository: UserRepository by lazy {
            UserRepository()
        }


        val gamesRepository: GamesRepository by lazy {
            GamesRepository()
        }

        val abstractCardRepository: AbstractCardRepository by lazy {
            AbstractCardRepository()
        }

        val wikiApiRepository: WikiApiRepository by lazy {
            Log.d(TAG_LOG,"wikiRepo")
            WikiApiRepositoryImpl()
        }


    }
}